src/lightning_classify/data/datamodule.py

```diff
@@ -0,0 +1,58 @@
+from typing import Any
+from pathlib import Path
+
+from torchvision.datasets import ImageFolder
+from torch.utils.data import DataLoader
+from lightning_classify.transforms import ImageTransform
+
+from lightning import LightningDataModule
+
+
+class OnepieceImageDataModule(LightningDataModule):
```

Comment:
i want you to use this as an example for standard code for data module

```python
class MNISTDataModule(L.LightningDataModule):
    def __init__(self, data_dir: str = "path/to/dir", batch_size: int = 32):
        super().__init__()
        self.data_dir = data_dir
        self.batch_size = batch_size

    def setup(self, stage: str):
        self.mnist_test = MNIST(self.data_dir, train=False)
        self.mnist_predict = MNIST(self.data_dir, train=False)
        mnist_full = MNIST(self.data_dir, train=True)
        self.mnist_train, self.mnist_val = random_split(
            mnist_full, [55000, 5000], generator=torch.Generator().manual_seed(42)
        )

    def train_dataloader(self):
        return DataLoader(self.mnist_train, batch_size=self.batch_size)

    def val_dataloader(self):
        return DataLoader(self.mnist_val, batch_size=self.batch_size)

    def test_dataloader(self):
        return DataLoader(self.mnist_test, batch_size=self.batch_size)

    def predict_dataloader(self):
        return DataLoader(self.mnist_predict, batch_size=self.batch_size)

    def teardown(self, stage: str):
        # Used to clean-up when the run is finished
        ...
        
   ```
   
   so from your code move trainset and validset  to setup
   
   ```python
   def setup(self, stage:str):
      self.trainset = self._build_dataset(mode="train")
      self.validset = self._build_dataset(mode="valid")

      self.class_names = self.trainset.classes
      self.class_to_idx = self.trainset.class_to_idx
        
    ```